src/eon.erl

```diff
@@ -233,6 +251,38 @@ zip(Obj1, Obj2) ->
                , lists:sort(new(Obj1)), lists:sort(new(Obj2)) ).
 
 %%%_ * Higher-order ----------------------------------------------------
+-spec all(func(boolean()), object(_, _)) -> boolean().
+%% @doc all(F, Obj) returns `true` if `F` evaluates to `true` for all
+%% entries in Obj.
+all(F, Obj) when is_function(F, 1) ->
+  foldl_while(fun(V, _Acc) -> format(F(V), ok_true) end, true, Obj);
+all(F, Obj) when is_function(F, 2) ->
+  foldl_while(fun(K, V, _Acc) -> format(F(K, V), ok_true) end, true, Obj).
+
+
+-spec any(func(boolean()), object(_, _)) -> boolean().
+%% @doc any(F, Obj) returns `true` if `F` evaluates to `true` for any
+%% entry in Obj.
+any(F, Obj) when is_function(F, 1) ->
+  foldl_while(fun(V, _Acc) -> format(F(V), ok_false) end, false, Obj);
+any(F, Obj) when is_function(F, 2) -> 
+  foldl_while(fun(K, V, _Acc) -> format(F(K, V), ok_false) end, false, Obj).
+
+
+foldl_while(F, Acc, [{_K, V} | Tail]) when is_function(F, 2) ->
+  continue_or_acc(F, F(V, Acc), Tail);
+foldl_while(F, Acc, [{K, V} | Tail]) when is_function(F, 3) ->
+  continue_or_acc(F, F(K, V, Acc), Tail);
+foldl_while(_F, Acc, []) -> Acc.
+
+continue_or_acc(F, {ok, Acc}, Obj) -> foldl_while(F, Acc, Obj);
+continue_or_acc(_, {stop, Acc}, _) -> Acc.
+
+format(true, ok_true)   -> {ok, true};
+format(false, ok_true)  -> {stop, false};
+format(false, ok_false) -> {ok, false};
+format(true, ok_false)  -> {stop, true}.
```

Comment:
I would suggest that it would be a good idea for readability to split format/2 into two functions. One that can be used by any/2 and one that can be used by all/2, then we would not need to use `ok_false` and `ok_true` which in my opinion makes the code a bit harder to read and understand.